maven-embedder/src/main/java/org/apache/maven/cli/internal/BootstrapCoreExtensionManager.java

```diff
@@ -77,24 +79,29 @@ public class BootstrapCoreExtensionManager {
 
     private final ClassRealm parentRealm;
 
+    private final WorkspaceReader ideWorkspaceReader;
+
     @Inject
     public BootstrapCoreExtensionManager(
             DefaultPluginDependenciesResolver pluginDependenciesResolver,
             DefaultRepositorySystemSessionFactory repositorySystemSessionFactory,
             CoreExports coreExports,
-            PlexusContainer container) {
+            PlexusContainer container,
+            @Nullable @Named("ide") WorkspaceReader ideWorkspaceReader) {
         this.pluginDependenciesResolver = pluginDependenciesResolver;
         this.repositorySystemSessionFactory = repositorySystemSessionFactory;
         this.coreExports = coreExports;
         this.classWorld = ((DefaultPlexusContainer) container).getClassWorld();
         this.parentRealm = container.getContainerRealm();
+        this.ideWorkspaceReader = ideWorkspaceReader;
     }
 
     public List<CoreExtensionEntry> loadCoreExtensions(
             MavenExecutionRequest request, Set<String> providedArtifacts, List<CoreExtension> extensions)
             throws Exception {
         try (CloseableSession repoSession = repositorySystemSessionFactory
                 .newRepositorySessionBuilder(request)
+                .setWorkspaceReader(ideWorkspaceReader)
```

Comment:
Won't that override the existing chained workspace reader ? is that the intent ?

Reply:
With this PR session factory does NOT set any workspace reader anymore (read PR description and see code), it is _caller duty_

Reply:
Yes, I don't question the change or responsibilities, but a possible change in behaviour.  
The caller is `BootstrapCoreExtensionManager` and now responsible to set the workspace reader on the session.  What I'm wondering is if this leads back to the same behaviour... I think before this PR, the workspace reader is to set a chained one, now, it's only the `ide` one...

Reply:
Oh, you mean IF request has one as well? Hm, let me recheck is that possible...

Reply:
yes, clearly IS possible (as copy of "origina" request is used), fixing...

Reply:
Fixed.

But your comment is wrong: session factory _never used_ chained WSReader, it was ONLY the default maven using it. Before this PR, the session could be inited with request WS (if non null) OR the ide WS (that may be null).

Reply:
Fixed and also simplified, now chained WS omits possible null entries in collection, thus all these become one liner.